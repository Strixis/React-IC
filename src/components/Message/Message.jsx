import './Message.less';

import React, { Component } from 'react';
import PropType from 'prop-types';
import classNames from 'classnames';

class Message extends Component {
  static propType = {
    author: PropType.string,
    text: PropType.string,
  }

  render() {
    const { author, text } = this.props;

    const classes = classNames('message', {
      'message_owner': author !== 'Bot',
      'message_companion': author === 'Bot'
    })

    return (
    <li className={ classes }>
      <span className="message_author">{ author }</span>
      <span className="message_text">{ text }</span>
    </li>
    )
  }
};

export { Message };
