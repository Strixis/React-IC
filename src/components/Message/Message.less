@color: aliceblue;

.message {
    background-color: @color;
}
