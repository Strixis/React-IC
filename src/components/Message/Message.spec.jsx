import React from 'react';
import renderer from 'react-test-renderer';

import { Message } from './Message';

test('Message should be required', () => {
  const tree = renderer.create(<Message
    author={ "Test Author" }
    message={ "It is a test message with some text" } />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
