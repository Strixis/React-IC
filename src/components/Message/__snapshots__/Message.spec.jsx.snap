// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should be required 1`] = `
<li
  className="message message_owner"
>
  <span
    className="message_author"
  >
    Test Author
  </span>
  <span
    className="message_text"
  />
</li>
`;
